loopy/target/execution.py

```diff
@@ -360,12 +360,22 @@ def handle_alloc(self, gen, arg, kernel_arg, strify, skip_arg_checks):
     def get_arg_pass(self, arg):
         raise NotImplementedError()
 
-    def get_strides_check_expr(self, shape, strides, sym_strides):
+    def get_strides_check_expr(self, shape, strides, expected_strides):
+        assert len(shape) == len(strides) == len(expected_strides)
+
         # Returns an expression suitable for use for checking the strides of an
         # argument. Arguments should be sequences of strings.
-        return " and ".join(
-                "(%s == 1 or %s == %s)" % elem
-                for elem in zip(shape, strides, sym_strides)) or "True"
+        match_expr = " and ".join(
+                f"({shape_i} == 1 or {strides_i} == {expected_strides_i})"
+                for shape_i, strides_i, expected_strides_i
+                in zip(shape, strides, expected_strides)) or "True"
+
+        for shape_i in shape:
+            # If any shape component is zero, the array is empty and the strides
+            # don't matter.
+            match_expr += f" or not {shape_i}"
```

Comment:
This relies on `and` binding more tightly than `or`: https://docs.python.org/3/reference/expressions.html#operator-precedence

Should we make this more explicit with parentheses?

Reply:
Yep, would be helpful. Could you also add a comment explaining why a 1-long axis is treated specially?

Reply:
Both done in 65ee212be4fafc0dbf8af0e5ed16b14bac37720b.